features/storage/kvstore/filesystemstore/FileSystemStore.h

```diff
@@ -0,0 +1,229 @@
+     *
+     * @param[in]  key                  Key.
+     * @param[out] info                 Returned information structure.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int get_info(const char *key, info_t *info);
+
+    /**
+     * @brief Remove a FileSystemStore item, given key.
+     *
+     * @param[in]  key                  Key.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int remove(const char *key);
+
+    /**
+     * @brief Start an incremental FileSystemStore set sequence.
+     *
+     * @param[out] handle               Returned incremental set handle.
+     * @param[in]  key                  Key.
+     * @param[in]  final_data_size      Final value data size.
+     * @param[in]  create_flags         Flag mask.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_start(set_handle_t *handle, const char *key, size_t final_data_size, uint32_t create_flags);
+
+    /**
+     * @brief Add data to incremental FileSystemStore set sequence.
+     *
+     * @param[in]  handle               Incremental set handle.
+     * @param[in]  value_data           value data to add.
+     * @param[in]  data_size            value data size.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_add_data(set_handle_t handle, const void *value_data, size_t data_size);
+
+    /**
+     * @brief Finalize an incremental FileSystemStore set sequence.
+     *
+     * @param[in]  handle               Incremental set handle.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_finalize(set_handle_t handle);
+
+    /**
+     * @brief Start an iteration over FileSystemStore keys.
+     *
+     * @param[out] it                   Returned iterator handle.
+     * @param[in]  prefix               Key prefix (null for all keys).
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_open(iterator_t *it, const char *prefix = NULL);
+
+    /**
+     * @brief Get next key in iteration.
+     *
+     * @param[in]  it                   Iterator handle.
+     * @param[in]  key                  Buffer for returned key.
+     * @param[in]  key_size             Key buffer size.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_next(iterator_t it, char *key, size_t key_size);
+
+    /**
+     * @brief Close iteration.
+     *
+     * @param[in]  it                   Iterator handle.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_close(iterator_t it);
+
+private:
+
+    /**
+     * Build Full name of Key as a combination of FSST folder and key name
+     *
+     * @returns 0 on success or a negative error code on failure
```

Comment:
Please add Doxy parameter description for private member functions as well.

Reply:
Thanks, added doxy info